unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs

```diff
@@ -15,11 +16,16 @@ public class ExploreV2MenuComponentController : IExploreV2MenuComponentControlle
     internal IExploreV2MenuComponentView view;
     internal IPlacesAndEventsSectionComponentController placesAndEventsSectionController;
     internal InputAction_Trigger toggleExploreTrigger;
+    internal IExploreV2Analytics exploreV2Analytics;
+    internal ExploreSection currentOpenSection;
+    internal float lastTimeSectionWasOpen = 0f;
+    internal bool anyActionExecutedFromLastOpen = false;
 
     public void Initialize()
     {
+        exploreV2Analytics = CreateAnalyticsController();
         view = CreateView();
-        SetVisibility(false);
+        SetVisibility(false, false);
```

Comment:
As I stated previously, we should try as much as possible to keep the production code away from the analytics (besides sending events).

Take a look at this file in this PR: https://github.com/decentraland/unity-renderer/blob/432db71e9418ddd4ced724b1e60a6b77e923a981/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Shortcuts/ShortcutsController.cs#L57

To see how the tracking of the shortcut has been handled. We could move the analytic from within `SetVisibility` to the trigger action itself (in the listener below).

Reply:
I have refactorized the code in order to avoid to have analytics-related code and I think it is much better and clearer now :) Please, take a look, thanks!